Kickstarter-iOS/Views/Controllers/BackingViewController.swift

```diff
@@ -178,6 +217,13 @@ internal final class BackingViewController: UIViewController {
   }
 
   fileprivate func goToMessages(project: Project, backing: Backing) {
+    if let nav = self.navigationController {
```

Comment:
I got a bit lost here with the `for` loop and `return` and all. What if we did something like:

``` swift
guard let nav = self.navigationController else { return }

if nav.childViewControllers.contains(where: { $0 is MessagesViewController }) {
  nav.popViewController(animated: true)
  return
}

let vc = MessagesViewController.configuredWith(project: project, backing: backing)
nav.pushViewController(vc, animated: true)
```